rules/group.go

```diff
@@ -214,32 +310,7 @@ func (g *Group) run(ctx context.Context) {
 		}(time.Now())
 	}()
 
-	g.evalIterationFunc(ctx, g, evalTimestamp)
-	if g.shouldRestore {
```

Comment:
One consequence of removing this one, is that we restore only after SECOND eval after time ticks. I think that's fine,given commentary... We do exactly that 🤔 Curious why we didn't do that in the first place!